trie/sync.go

```diff
@@ -344,7 +377,7 @@ func (s *Sync) children(req *request, object node) ([]*request, error) {
 		// Notify any external watcher of a new key/value node
 		if req.callback != nil {
 			if node, ok := (child.node).(valueNode); ok {
-				if err := req.callback(req.path, node, req.hash); err != nil {
+				if err := req.callback(child.path, node, req.hash); err != nil {
```

Comment:
This change looks like it's pretty substantial -- like this was previously wrong. What's the impact of changing this here?

Reply:
It was completely wrong before. It didn't pass in the full path to the leaf, rather omitted the suffix of the short node that ends in the leaf.

Reply:
So if we had a path like:

FullNode --0--> FullNode --1--> ShortNode --23456--> Account

Then the callback used the path `01` instead of `0123456` because the delivered request was the ShortNode, but we're calling the callback on the child of the short node.

Reply:
And now, does it use `23456` or `0123456` ?

Reply:
https://github.com/ethereum/go-ethereum/blob/master/trie/sync.go#L327